core/keyboard_nav/ast_node.ts

```diff
@@ -407,9 +456,14 @@ export class ASTNode {
    */
   next(): ASTNode | null {
     switch (this.type) {
-      case ASTNode.types.STACK:
-        return this.navigateBetweenStacks(true);
-
+      case ASTNode.types.STACK: {
+        const block = this.location as Block;
+        if (block.workspace.isFlyout) {
+          return this.navigateFlyoutContents(true);
+        } else {
+          return this.navigateBetweenStacks(true);
+        }
+      }
```

Comment:
I don't think subclassing is a requirement right now, but I would suggest moving this check into the `navigateBetweenStacks` method rather than repeating this check in multiple places.

Reply:
Moved the `.isFlyout` check into `navigateBetweenStacks`, as @maribethb suggested.